### PR TITLE
Add expected failing test for path with glob

### PR DESCRIPTION
**Pull request by bhack**
Add a failing test for https://github.com/tensorflow/tensorflow/issues/35489

/cc @mihaimaruseac 

You can test locally with:
`bazel test --test_arg=FileIoTest.testGetMatchingFilesWhenContainsGlob  //tensorflow/python/lib/io:file_io_test`

When the test will fail it is probably fixed and you can remove `@unittest.expectedFailure` to pass the test. 
You could temp comment `#@unittest.expectedFailure` for debugging the failure when you are developing a fix.

When it will be fixed, more in general it is suggested that you will locally test the whole target:
`bazel test --test_arg=FileIoTest.testGetMatchingFilesWhenContainsGlob  //tensorflow/python/lib/io:file_io_test`

### CODE REVIEW

**Comment by bhack** [comment]
@gbaned I don't think It could be classfied as `prtype:bugfix`. 
@mihaimaruseac Probably we need to creare a new label.

File: tensorflow/python/lib/io/file_io_test.py [hunk 1]
```diff
@@ -195,6 +196,20 @@ def testGetMatchingFilesWhenParentDirContainsParantheses(self):
     self.assertItemsEqual(
         file_io.get_matching_files(glob_pattern), expected_match)
 
+  @unittest.expectedFailure
+  def testGetMatchingFilesWhenContainsGlob(self):
+    # https://git.io/JR8jL (#35489)
```
Comment (mihaimaruseac) [inline tensorflow/python/lib/io/file_io_test.py:201]: Can we make this be `expectedFailure(link to github issue)`?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/lib/io/file_io_test.py (hunk 1 above)
Comment (bhack) [reply tensorflow/python/lib/io/file_io_test.py:201]: I don't it has a reason like argument
https://docs.python.org/3/library/unittest.html

File: tensorflow/python/lib/io/file_io_test.py [hunk 2]
```diff
@@ -195,6 +196,20 @@ def testGetMatchingFilesWhenParentDirContainsParantheses(self):
...
+    dir_path = os.path.join(self._base_dir, "dir_[special]")
+    file_io.create_dir(dir_path)
+    files = ["file1.txt", "file*2?].txt"]
+    for name in files:
+      file_path = os.path.join(dir_path, name)
+      file_io.FileIO(file_path, mode="w").write("testing")
+    expected_match = [os.path.join(dir_path, name) for name in files]
+    glob_pattern = os.path.join(dir_path, "*")
+    self.assertItemsEqual(
```
Comment (bhack) [inline tensorflow/python/lib/io/file_io_test.py:210]: We could evaluate the `msg` arg here

**Comment by bhack** [comment]
Check if you like with a message when it fail (so when it is probably fixed).

File: tensorflow/python/lib/io/file_io_test.py [hunk 3]
```diff
@@ -195,6 +196,20 @@ def testGetMatchingFilesWhenParentDirContainsParantheses(self):
...
+        file_io.get_matching_files(glob_pattern), expected_match,
+        "Check #35489 https://git.io/JR8jL")
```
Comment (mihaimaruseac) [inline tensorflow/python/lib/io/file_io_test.py:211]: Let's change this to be the direct link to the issue (e.g. `https://github.com/tensorflow/tensorflow/issue/35489`)

**Review by mihaimaruseac — COMMENTED** [review]
Looks good, minor nit.

Need to also check if this would internally.

File: tensorflow/python/lib/io/file_io_test.py (hunk 3 above)
Comment (bhack) [reply tensorflow/python/lib/io/file_io_test.py]: Done

**Review by mihaimaruseac — APPROVED** [review]

**Comment by bhack** [comment]
@mihaimaruseac We had an `import/copybara` fail, let me know tomorrow

**Comment by gbaned** [comment]
Here are the internal errors, @bhack  can you please verify ?

Traceback (most recent call last):
  File "/tensorflow/python/lib/io/file_io_test.py", line 59, in tearDown
    file_io.delete_recursively(self._base_dir)
  File "/tensorflow/python/lib/io/file_io.py", line 664, in delete_recursively
    delete_recursively_v2(dirname)
  File "/tensorflow/python/lib/io/file_io.py", line 677, in delete_recursively_v2
    _pywrap_file_io.DeleteRecursively(compat.path_to_bytes(path))
google3.third_party.tensorflow.python.framework.errors_impl.FailedPreconditionError: Delete failed: /tmp/tmprxj1885g/base_dir/[dir_special]: Directory not empty

**Comment by bhack** [comment]
@gbaned I haven't visibility about these internal tests. We need to wait for @mihaimaruseac

**Comment by bhack** [comment]
@mihaimaruseac 

Can I wrap this as a conditional expected fail for the copybara internal env (if is it still in the oss sourcecode)?

```
_pywrap_file_io.DeleteRecursively(compat.path_to_bytes(path))
google3.third_party.tensorflow.python.framework.errors_impl.FailedPreconditionError: Delete failed: /tmp/tmprxj1885g/base_dir/[dir_special]: Directory not empty
```

**Review by mihaimaruseac — APPROVED** [review]

**Comment by bhack** [comment]
Thank you for running this but it is failing again internally. 
I will close this as we don't have a clear position and allocated resources about the expecting failing tests proposal.

I will open some of these PRs again in the case this process it will be officially approved.

**Pull request closed without merging**